networklibrary/src/main/java/com/eis/smsnetwork/smsnetcommands/SMSQuitNetwork.java

```diff
@@ -0,0 +1,39 @@
+package com.eis.smsnetwork.smsnetcommands;
+
+import androidx.annotation.NonNull;
+
+import com.eis.communication.network.NetSubscriberList;
+import com.eis.communication.network.commands.QuitNetwork;
+import com.eis.smslibrary.SMSPeer;
+import com.eis.smsnetwork.RequestType;
+import com.eis.smsnetwork.SMSJoinableNetManager;
+import com.eis.smsnetwork.broadcast.BroadcastSender;
+
+/**
+ * Command to quit the current network.
+ *
+ * @author Edoardo Raimondi
+ * @author Marco Cognolato
+ * @author Giovanni Velludo
+ */
+public class SMSQuitNetwork extends QuitNetwork<SMSPeer> {
+
+    /**
+     * Constructor for the SMSQuitNetwork command, needs the data to operate
+     *
+     * @param netSubscribers The subscribers currently in the network
+     */
+    public SMSQuitNetwork(@NonNull NetSubscriberList<SMSPeer> netSubscribers) {
+        super(netSubscribers);
+    }
+
+    /**
+     * Removes myself from the subscribers list and broadcasts it to the net
+     */
+    protected void execute() {
+        String quitNetworkMessage = RequestType.QuitNetwork.asString();
+        BroadcastSender.broadcastMessage(netSubscribers.getSubscribers(), quitNetworkMessage);
+        SMSJoinableNetManager.getInstance().getNetSubscriberList().getSubscribers().clear();
```

Comment:
A new method `SMSJoinableNetManager.clear()` might be useful for this command.

Reply:
How about also asking for the SMSJoinableNetManager in the constructor? lol